### PR TITLE
RTM: CreateAndStartLocalNet func optimization

### PR DESCRIPTION
**Pull request by egorka-sereda**
move reconnecting mechanism inside CreateAndStartLocalNet func
remove loggerFactory() from create/start network functions
add CI workflows launching also at PRs

### CODE REVIEW

File: network/p2p/test_utils.go [hunk 1]
```diff
@@ -69,44 +66,59 @@ func (ln *LocalNet) WithBootnode(ctx context.Context, logger *zap.Logger) error
...
+
+		eg, ctx := errgroup.WithContext(pCtx)
+		for i, node := range ln.Nodes {
+			i, node := i, node //hack to avoid closures. price of using error groups
+
+			eg.Go(func() error {
+				if err := node.Start(); err != nil {
+					return fmt.Errorf("could not start node %d: %w", i, err)
+				}
+				ctx, cancel := context.WithTimeout(ctx, 1000*time.Second)
```
Comment (nkryuchkov) [inline network/p2p/test_utils.go:86]: Is `1000*time.Second` intentional? It's 16.7 minutes

Comment (egorka-sereda) [reply network/p2p/test_utils.go]: it's a mistake, fixed

**Review by nkryuchkov — APPROVED** [review]

**Review by nivBlox — COMMENTED** [review]
please make sure integration test are passing

**Review by nivBlox — APPROVED** [review]

**Comment by egorka-sereda** [comment]
**Ready to merge!**

**Comment by aaron-blox** [comment]
is this PR still relevant @egorka-sereda ?

**Comment by egorka-sereda** [comment]
these changes already was merged at this PR
https://github.com/bloxapp/ssv/pull/830

**Pull request closed without merging**